src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java

```diff
@@ -151,12 +153,22 @@ private void onRemoval(RemovalNotification<String, KNNIndexCacheEntry> removalNo
      */
     public KNNIndex getIndex(String key, final String indexName) {
         try {
+            //TODO if Type Not consistent
```

Comment:
What do you mean by this?

Reply:
when cached an Index, should we verify which engine for this index, and then load the exactly library?

Reply:
I see. I think we should add a different file extension for faiss graphs as opposed to nmslib graphs. Maybe <filename>.faiss_hnsw.